### ISSUE TITLE
LDAP Authentication should be independent of registration setting

### ISSUE THREAD

**Issue by Rhywden** [bigbluebutton/greenlight#1741]
The issue is the following: Thanks to the latest update the users of my school can now successfully login to our instance. Thanks for the speedy fix!

There's one snag, however: The registrations setting still needs to be "Open" because otherwise they either cannot login in at all (Invitation - new LDAP users will get the "You need an invite!" message) or I'd have to approve every single one of them (Approve/Deny). With 1700+ potential users you can guess that the latter is not a method I'd like to do.

Thus I have to set the registration setting to "Open" and hope that we don't have a random troll wandering in - while I've set the normal user capabilities to zero, that does not prevent them from joining an open room (though I've set the default to "Moderator needs to approve new attendees" and "Meeting only starts with a moderator")

Basically, the system needs to treat LDAP users as independent from any registration setting - after all, they're pre-approved and their role is set through the LDAP binding.

As a further enhancement I'd also like the setting "No external registrations" - it simply is not needed in our use case.
Oh, and I'd also like the option to disable recordings globally.

**Comment by mrkeksi** [comment]
If I understand it correctly, you simply mean that unregistered people cannot participate in any conference?
What is with the setting in the site settings:
![image](https://user-images.githubusercontent.com/8763656/83692764-e45b1880-a5f4-11ea-815f-4965415e0a6d.png)

**Comment by Rhywden** [comment]
That will not do anything because the **registration** must be **open** or I'll have a pretty LDAP sign-in my users cannot actually use.

**Comment by ichdasich** [comment]
You can disable greenlight accounts in .env:

set `ALLOW_GREENLIGHT_ACCOUNTS=false`

**Comment by Rhywden** [comment]
Well, yes. But then we hit another problem: We only have two roles in our LDAP repo - teachers and students. If I give the "admin" role to someone who logs in through LDAP he'll automatically revert to a "teacher" role the next time he/she logs in.

**Comment by farhatahmad** [comment]
I'm a little bit confused as to what the problem actually is. Setting `ALLOW_GREENLIGHT_ACCOUNTS=false` means that only your LDAP users can sign in which solves your problem. 

> If I give the "admin" role to someone who logs in through LDAP he'll automatically revert to a "teacher" role the next time he/she logs in.

This is expected. It's designed to always match the role given from LDAP. You have 2 options here:
Are all teachers admins? or only some? If all, then just give the teacher role the same permissions as the admin role. If only some, you'll probably need to make custom changes so that it only sets the role if the user doesn't already have one

**Comment by Rhywden** [comment]
The "usual" way this is done is to decouple the login systems. So that you have a fallback in case the LDAP server is down.

**Comment by farhatahmad** [comment]
> The "usual" way this is done is to decouple the login systems.

Can you expand a bit more on this?

**Comment by Rhywden** [comment]
Well, you usually have "trusted" sources of login/registration (like LDAP, SSO or a custom OAuth provider) and "untrusted" sources of login/registration (Facebook/GMail/... Oauth, Registration by Email,...)
The trusted sources are pre-approved because you assume that they're only delivering valid accounts.

This here is from our school's Moodle admin page - you can see the various registration sources and they're all independent of each other. 

![image](https://user-images.githubusercontent.com/84586/83773331-ae657500-a684-11ea-8a54-c5926399112c.png)

I

**Comment by farhatahmad** [comment]
Oh I see what you mean now. There is already a feature request (#1020) for allowing each authentication method to have it's own registration method, so I'm going to close this as a duplicate
